pkg/manager/member/tikv_upgrader.go

```diff
@@ -174,6 +175,10 @@ func (u *tikvUpgrader) Upgrade(meta metav1.Object, oldSet *apps.StatefulSet, new
 			continue
 		}
 
+		if isStable, reason := pdapi.IsClusterStable(controller.GetPDClient(u.deps.PDControl, tc)); !isStable {
+			return controller.RequeueErrorf("cluster is unstable: %s", reason)
+		}
+
```

Comment:
Isn't the cluster stability already checked above? Probably it's a duplication that we can remove?
https://github.com/pingcap/tidb-operator/blob/7a9fe4126e81691796320cba0cf7115f77c22f49/pkg/manager/member/tikv_upgrader.go#L74

Reply:
Oh, I meant, in method `isTiKVReadyToUpgrade` we have already checked whether the TiKV nodes are stable
https://github.com/pingcap/tidb-operator/blob/7a9fe4126e81691796320cba0cf7115f77c22f49/pkg/manager/member/tikv_upgrader.go#L510
and as `isTiKVReadyToUpgrade` is invoked above, do we need to double-check the TiKV stability here? Or we can keep the check here but remove the check in `isTiKVReadyToUpgrade`, and leave it as it used to be. What do you think?